packages/gestalt/src/Box.js

```diff
@@ -80,96 +54,818 @@ extra runtime typechecks in the transformers for performance.
 
 */
 
+type UnionRefs = HTMLDivElement | HTMLAnchorElement;
+
+// These types are mostly defined in boxTypes.js. Unfortunately to get generated docs
+// to work, we had to copy those types here. Ideally we can undo that copying in the
+// future if we figure out how to get docgen to resolve types defined elsewhere. =(
+
 // Please update `eslint-plugin-gestalt/no-box-disallowed-props` if you make changes to these props
 type Props = {
+  /**
+   *
+   */
   children?: Node,
-  dangerouslySetInlineStyle?: DangerouslySetInlineStyle,
-
-  display?: Display,
-  column?: Column,
-  direction?: Direction,
-  smDisplay?: Display,
-  smColumn?: Column,
-  smDirection?: Direction,
-  mdDisplay?: Display,
-  mdColumn?: Column,
-  mdDirection?: Direction,
-  lgDisplay?: Display,
-  lgColumn?: Column,
-  lgDirection?: Direction,
-
-  alignContent?: AlignContent,
-  alignItems?: AlignItems,
-  alignSelf?: AlignSelf,
-  as?: As,
-  bottom?: Bottom,
-  borderStyle?: BorderStyle,
-  color?: Color,
-  fit?: Fit,
-  flex?: Flex,
-  height?: Dimension,
-  justifyContent?: JustifyContent,
-  left?: Left,
-
-  margin?: Margin,
-  marginTop?: Margin,
-  marginBottom?: Margin,
-  marginStart?: Margin,
-  marginEnd?: Margin,
-
-  smMargin?: Margin,
-  smMarginTop?: Margin,
-  smMarginBottom?: Margin,
-  smMarginStart?: Margin,
-  smMarginEnd?: Margin,
-
-  mdMargin?: Margin,
-  mdMarginTop?: Margin,
-  mdMarginBottom?: Margin,
-  mdMarginStart?: Margin,
-  mdMarginEnd?: Margin,
-
-  lgMargin?: Margin,
-  lgMarginTop?: Margin,
-  lgMarginBottom?: Margin,
-  lgMarginStart?: Margin,
-  lgMarginEnd?: Margin,
-
-  maxHeight?: Dimension,
-  maxWidth?: Dimension,
-  minHeight?: Dimension,
-  minWidth?: Dimension,
-
-  opacity?: Opacity,
-
-  overflow?: Overflow,
-
-  padding?: Padding,
-  smPadding?: Padding,
-  mdPadding?: Padding,
-  lgPadding?: Padding,
-
-  paddingX?: Padding,
-  smPaddingX?: Padding,
-  mdPaddingX?: Padding,
-  lgPaddingX?: Padding,
-
-  paddingY?: Padding,
-  smPaddingY?: Padding,
-  mdPaddingY?: Padding,
-  lgPaddingY?: Padding,
-
-  position?: Position,
-  right?: Right,
-  rounding?: Rounding,
-  top?: Top,
-  width?: Dimension,
-  wrap?: Wrap,
-
-  userSelect?: UserSelect,
-
-  role?: Role,
-
+  /**
+   * An "escape hatch" used to apply styles not otherwise available on Box.
+   */
+  dangerouslySetInlineStyle?: {|
+    __style: { [key: string]: string | number | void },
+  |},
+
+  /**
+   * Aligns a flex container's lines within when there is extra space in the cross-axis, similar to how justify-content aligns individual items within the main-axis.
+   * Learn more about Flexbox layouts on [MDN](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox).
+   * Default: 'stretch'
```

Comment:
Since these default values are largely browser defaults or set in CSS, we need to use this feature to define them. This mostly applies to Box and Flex.